### PR TITLE
Ga 97 update post org add user

### PR DESCRIPTION
**Pull request by sanjayparekh-hmcts**
### Jira link (if applicable)
https://tools.hmcts.net/jira/browse/GA-97

### Change description ###
based off https://tools.hmcts.net/jira/browse/GA-22

### Checklist

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)
- [ ] Does this PR introduce a breaking change

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/professionalapi/controller/SuperController.java [hunk 1]
```diff
@@ -433,11 +429,16 @@ protected ResponseEntity<Object> inviteUserToOrganisation(NewUserCreationRequest
         var roles = newUserCreationRequest.getRoles();
         var professionalUser = validateInviteUserRequestAndCreateNewUserObject(newUserCreationRequest,
                 removeEmptySpaces(organisationIdentifier), roles);
+        ResponseEntity<Object> inviteRespnse;
         if (newUserCreationRequest.isResendInvite() && resendInviteEnabled) {
-            return reInviteExpiredUser(newUserCreationRequest, professionalUser, roles, organisationIdentifier);
+            inviteRespnse = reInviteExpiredUser(newUserCreationRequest, professionalUser, roles,
```
Comment (abul-syed) [inline src/main/java/uk/gov/hmcts/reform/professionalapi/controller/SuperController.java:434]:
```suggestion
            inviteResponse = reInviteExpiredUser(newUserCreationRequest, professionalUser, roles,
```

**Review by abul-syed — APPROVED** [review]
very small typo, other than that LGTM

File: src/main/java/uk/gov/hmcts/reform/professionalapi/service/impl/ProfessionalUserServiceImpl.java [hunk 2]
```diff
@@ -246,4 +271,57 @@ public void checkUserStatusIsActiveByUserId(String userId) {
...
+        } catch (Exception ex) {
+            throw new ExternalApiException(HttpStatus.valueOf(500), ERROR_USER_CONFIGURED_DELETE);
+        }
+
+        saveAllUserAccessTypes(professionalUser, userProfileUpdatedData.getUserAccessTypes());
+    }
+
+    private UserConfiguredAccess mapToUserConfiguredAccess(ProfessionalUser professionalUser,
+                                                           UserAccessType userAccessType) {
+        UserConfiguredAccess uca = new UserConfiguredAccess();
```
Comment (helenbirdhmcts) [inline src/main/java/uk/gov/hmcts/reform/professionalapi/service/impl/ProfessionalUserServiceImpl.java:315]: not sure about the name uca is userConfiguredAccess better?

File: src/main/java/uk/gov/hmcts/reform/professionalapi/util/RefDataUtil.java [hunk 3]
```diff
@@ -359,7 +359,7 @@ public static NewUserResponse findUserProfileStatusByEmail(String emailAddress,
                 if (errorResponse != null) {
                     log.error("{}:: Response from UserProfileByEmail service call {}",
                         loggingComponentName, errorResponse.getErrorDescription());
-                }                
+                }
```
Comment (helenbirdhmcts) [inline src/main/java/uk/gov/hmcts/reform/professionalapi/util/RefDataUtil.java:362]: perhaps undo changes in this file as its just white space updates and not needed

**Review by helenbirdhmcts — DISMISSED** [review]

File: src/main/java/uk/gov/hmcts/reform/professionalapi/util/RefDataUtil.java (hunk 3 above)
Comment (sanjayparekh-hmcts) [reply src/main/java/uk/gov/hmcts/reform/professionalapi/util/RefDataUtil.java:362]: The base branch has this - ill remove from there

File: src/contractTest/java/uk/gov/hmcts/reform/professionalapi/provider/ProfessionalExternalUserControllerProviderTestConfiguration.java [hunk 4]
```diff
@@ -38,6 +39,12 @@ public class ProfessionalExternalUserControllerProviderTestConfiguration extends
     @MockBean
     ServiceAuthFilter serviceAuthFilter;
 
+    @MockBean
+    UserConfiguredAccessRepository userConfiguredAccessRepository;
```
Comment (kiran-yenigala-hmcts) [inline src/contractTest/java/uk/gov/hmcts/reform/professionalapi/provider/ProfessionalExternalUserControllerProviderTestConfiguration.java:43]: userConfiguredAccessRepository, where is this used in test cases?

File: src/contractTest/java/uk/gov/hmcts/reform/professionalapi/provider/ProfessionalExternalUserControllerProviderTest.java [hunk 5]
```diff
@@ -193,6 +205,73 @@ private ProfessionalUser setupInteractionsForProfessionalUser() throws JsonProce
...
+        roleAdditionResponse.setIdamStatusCode("200");
+        modifyUserRolesResponse.setRoleAdditionResponse(roleAdditionResponse);
+
+        String bodyModifyUserRoles = objectMapper.writeValueAsString(modifyUserRolesResponse);
+
+        when(userProfileFeignClientMock.modifyUserRoles(any(), any(), any())).thenReturn(Response.builder()
+                .request(mock(Request.class)).body(bodyModifyUserRoles, Charset.defaultCharset()).status(200).build());
+
+        List<UserConfiguredAccess> allUserConfiguredAccess = new ArrayList<>();
+        allUserConfiguredAccess.add(new UserConfiguredAccess());
```
Comment (rajkatla-hmcts) [inline src/contractTest/java/uk/gov/hmcts/reform/professionalapi/provider/ProfessionalExternalUserControllerProviderTest.java:267]: consumer test expecting this new json array field , but i don't see here :
‘userAccessTypes’: [
        {
          ‘jurisdictionId’: ‘jurisdictionId1’,
          ‘organisationProfileId’: ‘organisationProfileId1’,
          ‘accessTypeId’: ‘accessTypeId1’,
          ‘enabled’: true
        }
      ]

File: src/contractTest/java/uk/gov/hmcts/reform/professionalapi/provider/ProfessionalExternalUserControllerProviderTest.java [hunk 6]
```diff
@@ -193,6 +205,73 @@ private ProfessionalUser setupInteractionsForProfessionalUser() throws JsonProce
         return professionalUser;
     }
 
+    private ProfessionalUser setupInteractionsForProfessionalUserWithUserAccessTypes() throws JsonProcessingException {
```
Comment (rajkatla-hmcts) [inline src/contractTest/java/uk/gov/hmcts/reform/professionalapi/provider/ProfessionalExternalUserControllerProviderTest.java:208]: don't see new provider test for adding new user to an organisation

**Review by rajkatla-hmcts — CHANGES_REQUESTED** [review]

File: src/contractTest/java/uk/gov/hmcts/reform/professionalapi/provider/ProfessionalExternalUserControllerProviderTestConfiguration.java (hunk 4 above)
Comment (sanjayparekh-hmcts) [reply src/contractTest/java/uk/gov/hmcts/reform/professionalapi/provider/ProfessionalExternalUserControllerProviderTestConfiguration.java:43]: added to a new test

File: src/contractTest/java/uk/gov/hmcts/reform/professionalapi/provider/ProfessionalExternalUserControllerProviderTest.java (hunk 5 above)
Comment (sanjayparekh-hmcts) [reply src/contractTest/java/uk/gov/hmcts/reform/professionalapi/provider/ProfessionalExternalUserControllerProviderTest.java:267]: It is there, but keeps getting overwritten on the broker with each new xui build

File: src/contractTest/java/uk/gov/hmcts/reform/professionalapi/provider/ProfessionalExternalUserControllerProviderTest.java (hunk 6 above)
Comment (sanjayparekh-hmcts) [reply src/contractTest/java/uk/gov/hmcts/reform/professionalapi/provider/ProfessionalExternalUserControllerProviderTest.java:208]: I did not change anything about adding a new user on this pr

**Review by rajkatla-hmcts — APPROVED** [review]

**Review by rajkatla-hmcts — DISMISSED** [review]

**Review by rajkatla-hmcts — DISMISSED** [review]

**Review by kiran-yenigala-hmcts — DISMISSED** [review]
One minor comment, Approved

File: src/main/java/uk/gov/hmcts/reform/professionalapi/controller/request/NewUserCreationRequest.java [hunk 7]
```diff
@@ -19,19 +21,23 @@ public class NewUserCreationRequest {
...
+    private Set<UserAccessType> userAccessTypes;
 
     @JsonCreator
     public NewUserCreationRequest(
             @JsonProperty("firstName") String firstName,
             @JsonProperty("lastName") String lastName,
             @JsonProperty("email") String emailAddress,
             @JsonProperty("roles") List<String> roles,
-            @JsonProperty("resendInvite") boolean resendInvite) {
+            @JsonProperty("resendInvite") boolean resendInvite,
```
[the remark this answers is not in the document]
Comment (sanjayparekh-hmcts) [reply src/main/java/uk/gov/hmcts/reform/professionalapi/controller/request/NewUserCreationRequest.java]: fixed

**Review by kiran-yenigala-hmcts — DISMISSED** [review]

**Review by rajkatla-hmcts — APPROVED** [review]

**Review by bkpatel81 — DISMISSED** [review]

**Review by bkpatel81 — APPROVED** [review]

**Review by abul-syed — APPROVED** [review]

**Pull request merged**